[DataLoader] TransformsIterDataPipe

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52604 [DataLoader] TransformsIterDataPipe**

Add Transforms for `torchvideo.transfroms` and `torchaudio.transforms` with type checking.

Differential Revision: [D26581511](https://our.internmc.facebook.com/intern/diff/D26581511)

ejguan committed:
[DataLoader] TransformsIterDataPipe

[ghstack-poisoned]

VitalyFedyunin reviewed (approved):
With small pending fix

ejguan committed:
Update on "[DataLoader] TransformsIterDataPipe"


Add Transforms for `torchvideo.transfroms` and `torchaudio.transforms` with type checking.

Differential Revision: [D26581511](https://our.internmc.facebook.com/intern/diff/D26581511)

[ghstack-poisoned]

ejguan committed:
Update on "[DataLoader] TransformsIterDataPipe"


Add Transforms for `torchvideo.transfroms` and `torchaudio.transforms` with type checking.

Differential Revision: [D26581511](https://our.internmc.facebook.com/intern/diff/D26581511)

[ghstack-poisoned]
